tests/Integration/WhereConditions/WhereConditionsDirectiveTest.php

```diff
@@ -54,6 +56,116 @@ protected function getPackageProviders($app): array
         );
     }
 
+    public function testBetweenWithMultipleVariables(): void
+    {
+        $user1 = factory(User::class)->create();
+        assert($user1 instanceof User);
+        $user1->date_of_birth = Carbon::createStrict(2000, 1, 1);
+        $user1->save();
+
+        $user2 = factory(User::class)->create();
+        assert($user2 instanceof User);
+        $user2->date_of_birth = Carbon::createStrict(1995, 1, 1);
+        $user2->save();
+
+        $this->graphQL(/** @lang GraphQL */ '
+        query ($dobMin: Mixed, $dobMax: Mixed) {
+            users(
+                where: {
+                    column: "date_of_birth",
+                    operator: BETWEEN,
+                    value: [$dobMin, $dobMax]
+                }
+            ) {
+                id
+            }
+        }
+        ', [
+            'dobMin' => '1990-01-01',
+            'dobMax' => '1999-01-01',
+        ])->assertExactJson([
+            'data' => [
+                'users' => [
+                    [
+                        'id' => "{$user2->id}",
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    public function testBetweenWithOneVariable(): void
```

Comment:
I have added this test case, which works. So the part that does not work is more closely related to using variables in a list literal of a non-list type.

Reply:
Found the bug, see https://github.com/mll-lab/graphql-php-scalars/pull/35.